[Meng] iP

CreditableX opened the pull request:
Sure! Here's an updated version of the description with the additional requirements:

---

### iP PR Update 🚀

This PR introduces several key improvements and bug fixes:

- **Improved**: Enhanced performance of the `Bob` bot.
- **Fixed**: Resolved issue with handling special characters.
- **Added**: New features for better user interaction.

#### Key Changes:

1. **Performance Optimization**: Refactored core algorithms to reduce latency.
2. **Bug Fixes**: Addressed edge cases that caused unexpected crashes.
3. **New Features**: Implemented additional commands for user convenience.

```python
def improved_function():
    # Optimized function for better performance
    result = process_data()
    return result
```

- [x] **Code Review**: Complete
- [ ] **Testing**: In Progress
- [ ] **Documentation**: Not Started

> _"The best way to predict the future is to invent it."_ — Alan Kay

For more details, refer to the [project documentation](https://example.com/docs).

Please ensure that you update your local **__environment__** with the latest changes. 

This is an  `inline code`.

---

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

CreditableX committed:
added level 0

CreditableX committed:
changed name

CreditableX committed:
user interaction and bye command

CreditableX committed:
added string list function

CreditableX committed:
checkmark done / not done feature

CreditableX committed:
created task subclasses

CreditableX committed:
Add support for tracking three types of tasks:

CreditableX committed:
updated script + removed testing sysouit lines

CreditableX committed:
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

CreditableX committed:
changed taskList to an ArrayList + added delete feature

CreditableX committed:
retroactive testing runtest

CreditableX committed:
added localsave and updated script testing

CreditableX committed:
added localsave and updated script testing

CreditableX committed:
Merge branch 'branch-Level-7'
Save the tasks in the hard disk automatically whenever the task list changes.

CreditableX committed:
javadoc for updateDataFile method

CreditableX committed:
javadoc for updateDataFile method

CreditableX committed:
Teach the chatbot how to understand dates and times

CreditableX committed:
Merge branch 'branch-Level-8'
Teach the chatbot how to understand dates and times.

CreditableX committed:
OOP refactor

CreditableX committed:
Merge branch 'branch-A-MoreOOP'

OOP Refactor

CreditableX committed:
divide classes into packages

CreditableX committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Add gradle support

CreditableX committed:
deleted data file

CreditableX committed:
follow convention

CreditableX committed:
gradle setup

CreditableX committed:
add junit tests to parser, ui

CreditableX committed:
package app as jar

CreditableX committed:
add javadoc

CreditableX committed:
Refactor code to comply with coding standard

CreditableX committed:
Refactor code to comply with coding standard

CreditableX committed:
Refactor code to comply with coding standard

CreditableX committed:
Refactor code to comply with coding standard

CreditableX committed:
javadoc changes

CreditableX committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDocs.

CreditableX committed:
follow codingstandard

CreditableX committed:
Merge branch 'branch-A-CodingStandard'

follow codingstandard

CreditableX committed:
prelim level 9

CreditableX committed:
give users a way to find a task by searching for a keyword in the task description

CreditableX committed:
Merge branch 'branch-Level-9'

Give users a way to find a task by searching for a keyword in the task description.

CreditableX committed:
add comments

CreditableX committed:
add comments

CreditableX committed:
add comments

CreditableX committed:
add comments

CreditableX committed:
add comments

leslieyip02 commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,96 @@
+
+    public void run() throws CommandNotFoundException, MissingParamsException, PositionException {
+        // init items
+        boolean exit = false;
```
`exit` could be named as `shouldExit` so it sounds more like a boolean.

arnav-goel05 replied:
I agree, it could be potentially named `shouldExit` or `isExiting`.

leslieyip02 commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,96 @@
+    public void run() throws CommandNotFoundException, MissingParamsException, PositionException {
+        // init items
+        boolean exit = false;
+        String[] strArr = new String[2];
```
Maybe `strArr` can be renamed as `inputTokens` instead so it is clear that this is a multi-valued variable.

leslieyip02 commented on src/main/java/misc/Ui.java:
```diff
@@ -0,0 +1,159 @@
+    /**
+     * Return a blank line.
+     */
+    public String retBlank() {
+        return blankline;
+    }
```
Perhaps the `retBlank()` method can be renamed to `getBlankline()` for more clarity?

leslieyip02 commented on src/main/java/task/Tasklist.java:
```diff
@@ -0,0 +1,93 @@
+     * @param pos
+     * @return Task
+     */
+    public Task getStr(int pos) {
+        return taskList.get(pos);
+    }
```
I think this could be renamed as `getTask()` instead since it's returning a `Task` and not a `String`.

leslieyip02 commented on src/main/java/misc/Ui.java:
```diff
@@ -0,0 +1,159 @@
+public class Ui {
+    String blankline = "____________________________________________________________ \s";
+    
+    /**
+     * Prints a farewell message to the user.
+     */
+    public void endGame() {
+        System.out.println(blankline + "\n" + "Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Prints a start message.
+     */
+    public void startGame() {
+        System.out.println(  """
+            ____________________________________________________________ \s
+            Hello! I'm Bob \s
+            What can I do for you? \s
+            ____________________________________________________________ \s
+            """);
+    }
```
The naming of `startGame()` and `endGame()` might be a bit misleading since this isn't a game.

leslieyip02 commented on src/main/java/misc/Ui.java:
```diff
@@ -0,0 +1,159 @@
+     * Returns reply for event command.
+     */
+    public Tasklist replyEvent(String s, Tasklist tasklist) throws MissingParamsException {
+        String[] splitString = s.split("/from|/to", 3);
```
Similar to above, this could be renamed as something that is plural like `stringTokens` so it is clear that this is an array.

leslieyip02 reviewed (commented):
```
.____           __ /\          ________         __________             .__    .___      __   
|    |    _____/  |)/ ______  /  _____/  ____   \______   \ ___________|__| __| _/_____/  |_ 
|    |  _/ __ \   __\/  ___/ /   \  ___ /  _ \   |     ___// __ \_  __ \  |/ __ |/  _ \   __\
|    |__\  ___/|  |  \___ \  \    \_\  (  <_> )  |    |   \  ___/|  | \/  / /_/ (  <_> )  |  
|_______ \___  >__| /____  >  \______  /\____/   |____|    \___  >__|  |__\____ |\____/|__|  
        \/   \/          \/          \/                        \/              \/            
```

CreditableX committed:
add checkstyle

CreditableX committed:
change style

CreditableX committed:
Merge branch 'branch-A-CheckStyle'

Fix style.

arnav-goel05 commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,96 @@
+                    exit = true;
+                    ui.endGame();
+                    break;
+
```
Perhaps there's no need for the line break between each `case` statement.

arnav-goel05 commented on src/main/java/exceptions/PositionException.java:
```diff
@@ -0,0 +1,14 @@
+package exceptions;
+
+public class PositionException extends Exception {
+    private int pos;
```
Maybe you could utilize a more descriptive variable name such `position`.

arnav-goel05 commented on src/main/java/misc/Parser.java:
```diff
@@ -0,0 +1,27 @@
+        String input = scanner.nextLine();
+        String cmd = input.split(" ", 2)[0];
+        String rest = input.split(" ", 2).length > 1 ? input.split(" ", 2)[1] : "";
+        String[] retStrArr = new String[]{cmd, rest};
```
Perhaps it could be changed to `inputTokens` as stated above.

arnav-goel05 commented on src/main/java/misc/Ui.java:
```diff
@@ -0,0 +1,158 @@
+        }
+
+        try {
+            LocalDate d1 = LocalDate.parse(splitString[1].trim());
+            LocalDate d2 = LocalDate.parse(splitString[2].trim());
```
`d1` could be potentially renamed to `startDate` and `d2` could be renamed to `endDate` so that the variables are more descriptive.

arnav-goel05 reviewed (commented):
Looks great! Minor styling issues requested.

CreditableX committed:
update GUI

CreditableX committed:
update GUI

CreditableX committed:
change gradle

CreditableX committed:
restore exceptions

CreditableX committed:
restore exceptions

CreditableX committed:
Add assertions to various files

Assertions help to check for developer errors and are useful for early
debugging. Adding assertions gives me more layers of protection.

CreditableX committed:
improve code quality

CreditableX committed:
Merge pull request #2 from CreditableX/A-Assertions

add assertions

CreditableX committed:
Merge branch 'master' into A-CodeQuality

CreditableX committed:
Merge pull request #1 from CreditableX/A-CodeQuality

improve code quality

CreditableX committed:
add workflow

CreditableX committed:
Merge branch 'master' of https://github.com/CreditableX/ip

CreditableX committed:
add workflow

CreditableX committed:
add workflow

CreditableX committed:
add exact finding

CreditableX committed:
Delete src/main/data/dataFile.txt

CreditableX committed:
add javadocs

CreditableX committed:
add task tests

CreditableX committed:
add small changes

CreditableX committed:
Stop tracking .txt and .jar files

CreditableX committed:
Delete src/main/data/dataFile.txt

CreditableX committed:
Merge branch 'master' of https://github.com/CreditableX/ip

CreditableX committed:
change directory of datafile

CreditableX committed:
Update .gitignore to exclude data files

CreditableX committed:
Update .gitignore to exclude data files

CreditableX committed:
remove hardcode for datfile

CreditableX committed:
update storage

CreditableX committed:
change storage

CreditableX committed:
add small updates

CreditableX committed:
update workflow

CreditableX committed:
add gradle wrapper

CreditableX committed:
add javadocs

CreditableX committed:
update test

CreditableX committed:
fix bugs

CreditableX committed:
fix javadocs

CreditableX committed:
fix whitespaces
